src/kibana/components/vislib/lib/y_axis.js

```diff
@@ -48,6 +48,24 @@ define(function (require) {
     };
 
     /**
+     * By default, d3.format('s') returns billion values
+     * with a `G` instead of a `B`. @method formatAxisLabel returns
+     * billion values with a B instead of a G. Else, it defaults
+     * to the d3.format('s') value.
+     *
+     * @method formatAxisLabel
+     * @param d {Number}
+     * @returns {*}
+     */
+    YAxis.prototype.formatAxisLabel = function (d) {
+      var formatNumber = d3.format('s');
+      var str = formatNumber(d);
+      var replaceG = str.replace(/G/i, 'B');
+
```

Comment:
This seems like a weird way todo this. If D3 doesn't have a number formatter that can handle this maybe we should use a different formatting library. Numeral.js comes to mind.